tests/unit/test_kernels.py

```diff
@@ -1177,6 +1449,96 @@ def test_invalid_parameters(self, parameters: tuple, error_msg: str):
             SquaredExponentialKernel(*parameters)
 
 
+class TestMaternKernel(BaseKernelTest[MaternKernel], KernelMeanTest[MaternKernel]):
+    """Test ``coreax.kernels.MaternKernel``."""
+
+    @pytest.fixture(scope="class")
+    def kernel(self) -> MaternKernel:
+        random_seed = 2_024
+        parameters = jnp.abs(jr.normal(key=jr.key(random_seed), shape=(3,)))
+        return MaternKernel(
+            output_scale=parameters[0].item(),
+            length_scale=parameters[1].item(),
+            degree=int(jnp.ceil(jnp.abs(parameters[2]))) + 1,
+        )
+
+    @pytest.fixture(params=["floats", "vectors", "arrays"])
+    def problem(  # noqa: C901
+        self, request: pytest.FixtureRequest, kernel: MaternKernel
+    ) -> _Problem:
+        r"""
+        Test problems for the SquaredExponential kernel.
```

Comment:
This and the kernel defined below refer to squared exponential but should refer to Matern